### DISCUSSION TITLE
how to set background-color of icon?

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2130] [Q&A]
### Question

As we all know
```python
ui.icon('thumb_up', color='blue')
```
will draw an icon in blue color on default white background

what if a blue icon on red background? I have tried
```python
with ui.element('div').props("background-color='red'"):
    ui.icon('thumb_up', color='blue')
```
but it does not produce the red background. So how to fix the code? 

Furter, in fact, I want to use icon in ui.table, maybe something like
```python
table.add_slot('body-cell-num', '''
    <q-td key="num" :props="props">
        <q-icon size=1em :color="props.value < 21 ? 'blue' : 'white'" :background-color="props.value < 21 ? 'yellow' : 'blue'">
        </q-icon>
    </q-td>
''')
```

Thanks

**Comment by falkoschindler** [comment]
Unfortunately there is no background color prop in Quasar's QIcon: https://quasar.dev/vue-components/icon.
But you can either use a Tailwind or Quasar class or CSS:
```py
ui.icon('thumb_up', color='blue').classes('bg-red')  # Quasar class
ui.icon('thumb_up', color='blue').classes('bg-red-400')  # Tailwind class
ui.icon('thumb_up', color='blue').style('background-color: red')  # CSS style
```
